### PR TITLE
Implement Bottom Bar UI

### LINKED ISSUE #6: Implement Bottom bar UI
The bottom bar UI is ready. If the bottom bar UI in this Figma design `can't be implemented`, you can try a **`floating bottom bar UI type`**

Image

### PR DESCRIPTION
**Pull request by KMTus**
## Related issues
Resolve #6 

## Summary
Creat bottom bar all screen :  home , notification , favourite , account , oder
package : Add  library react-dom , d3-shape , animatable , curved-bottom-bar, type vecto icon

## Screenshots

Screenshot 2024-07-12 095158

### CODE REVIEW

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]
Why do you use the **`merge`** statement?

**Comment by KMTus** [comment]
> Why do you use the **`merge`** statemen
 If there is no defined process for using rebase, team members might misuse it, leading to inconsistencies in how commits are managed
Use Git Merge instead of rebase when working in groups: Git Merge keeps the commit history and is a safer choice when working in a group.

If it's wrong or bad activity, you can guide, pls

**Comment by KMTus** [comment]
> Why do you use the **`merge`** statement?

If there is no defined process for using rebase, team members might misuse it, leading to inconsistencies in how commits are managed
Use Git Merge instead of rebase when working in groups: Git Merge keeps the commit history and is a safer choice when working in a group.

If it's wrong or bad activity, you can guide, pls

**Comment by NguyenVanViet2004** [comment]
> > Why do you use the **`merge`** statement?
> 
> If there is no defined process for using rebase, team members might misuse it, leading to inconsistencies in how commits are managed Use Git Merge instead of rebase when working in groups: Git Merge keeps the commit history and is a safer choice when working in a group.
> 
> If it's wrong or bad activity, you can guide, pls

Reset commit and rebase `feature` branch from `main`

File: package.json [hunk 1]
```diff
@@ -49,11 +52,14 @@
...
     "react-i18next": "^14.1.0",
     "react-native": "0.74.2",
+    "react-native-animatable": "^1.4.0",
     "react-native-gesture-handler": "^2.16.2",
     "react-native-reanimated": "~3.10.1",
+    "react-native-redash": "^18.1.3",
     "react-native-safe-area-context": "4.10.1",
     "react-native-screens": "3.31.1",
     "react-native-svg": "15.2.0",
+    "react-native-vector-icons": "^10.1.0",
```
Comment (NguyenVanViet2004) [inline package.json:62]: Remove it

File: constants/Colors.ts [hunk 2]
```diff
@@ -27,8 +113,94 @@ const colors = {
...
+    primaryLite: '#637aff99',
+
+    purple: '#8f06e4',
+    purpleAlpha: 'rgba(146, 6, 228, 0.15)',
+    red: '#ff2f68',
+
+    redAlpha: 'rgba(255, 84, 84, 0.15)',
+    skyBlue: 'skyblue',
+    steelBlue: 'steelblue',
+
```
Comment (NguyenVanViet2004) [inline constants/Colors.ts:200]: Remove spacing

File: components/template/BottomTab.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+const TabButton: React.FC<TabButtonProps> =
+  ({ item, onPress, accessibilityState }) => {
+    const focused = accessibilityState?.selected ?? false
+    const viewRef = useRef<any>(null)
+    const circleRef = useRef<any>(null)
+    const textRef = useRef<any>(null)
+    const isDarkMode = useColorScheme() === 'dark'
+
+
+    const color = isDarkMode ? colors.white : colors.black
```
Comment (NguyenVanViet2004) [inline components/template/BottomTab.tsx:107]: Use getColors function

File: components/template/BottomTab.tsx [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        textRef.current?.transitionTo({ scale: 1 })
+      } else {
+        viewRef.current?.animate(animate2)
+        circleRef.current?.animate(circle2)
+        textRef.current?.transitionTo({ scale: 0 })
+      }
+    }, [focused])
+
+    return (
+      <TouchableOpacity
```
Comment (NguyenVanViet2004) [inline components/template/BottomTab.tsx:123]: Use the tamagui button if possible

File: components/atoms/Colors.tsx
Comment (NguyenVanViet2004) [inline components/atoms/Colors.tsx:1]: Remove this component, it is unnecessary

File: components/template/AccountScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { SafeAreaView, StyleSheet } from 'react-native'
+import { Text } from 'tamagui'
+const AccountScreen = (): any => {
```
Comment (NguyenVanViet2004) [inline components/template/AccountScreen.tsx:4]: `any` ?

File: components/template/BottomTab.tsx [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+  SafeAreaView,
+  StyleSheet,
+  TouchableOpacity,
+  useColorScheme,
+  View
+} from 'react-native'
+import * as Animatable from 'react-native-animatable'
+
+import getColors from '~/constants/Colors'
+import Icon, { Icons } from '../atoms/Icons'
```
Comment (NguyenVanViet2004) [inline components/template/BottomTab.tsx:17]: Use relative path

File: tsconfig.json
Comment (NguyenVanViet2004) [inline tsconfig.json:1]: Turn off automatic formatting codes when saving files

**Comment by NguyenVanViet2004** [comment]
image


Simulator Screenshot - iPhone 15 Pro Max - 2024-07-17 at 17 13 53


Ảnh màn hình 2024-07-17 lúc 17 14 30

**Comment by NguyenVanViet2004** [comment]
Fix it

**Comment by KMTus** [comment]
> Fix it

I used Dimensions to Responsive Text Bottom Bar
Hook errors, I don't use them nor create them, give solutions
image
The image appears is not good  
@thanhdo29 You can correct the functions and I see the problem of the greeting screen

File: components/templates/FavoriteScreen.tsx
Comment (NguyenVanViet2004) [inline components/templates/FavoriteScreen.tsx:1]: Could you change the name of this component from 'FavoriteScreen' to 'FavoriteTemplate'?

File: app/BottomBar.tsx [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+        textRef.current?.transitionTo({ scale: 1 })
+      } else {
+        viewRef.current?.animate(animate2)
+        circleRef.current?.animate(circle2)
+        textRef.current?.transitionTo({ scale: 0 })
+      }
+    }, [focused])
+
+    return (
+      <Button
```
Comment (NguyenVanViet2004) [inline app/BottomBar.tsx:72]: You should add the `unstyled` property here

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**